lib/slack/web/faraday/response/raise_error.rb

```diff
@@ -3,28 +3,33 @@ module Slack
   module Web
     module Faraday
       module Response
-        class RaiseError < ::Faraday::Response::Middleware
+        class RaiseError < ::Faraday::Middleware
           def on_complete(env)
             raise Slack::Web::Api::Errors::TooManyRequestsError, env.response if env.status == 429
 
             return unless env.success?
+            return unless env.body
 
-            body = env.body
-            return unless body
+            body = process_body(env)
             return if body['ok']
 
-            error_message =
-              body['error'] || body['errors'].map { |message| message['error'] }.join(',')
-
+            error_message = body['error'] || body['errors'].map { |message| message['error'] }.join(',')
             error_class = Slack::Web::Api::Errors::ERROR_CLASSES[error_message]
             error_class ||= Slack::Web::Api::Errors::SlackError
             raise error_class.new(error_message, env.response)
+          # TODO: Do we need to still handle ::Faraday::ParsingError or will this be handled else where?
+          # rescue ::Faraday::ParsingError
+          rescue ::JSON::ParserError
+            raise Slack::Web::Api::Errors::ParsingError, env.response
           end
 
-          def call(env)
-            super
-          rescue ::Faraday::ParsingError
-            raise Slack::Web::Api::Errors::ParsingError, env.response
+          private
+
+          # Need to check if the body has been parsed here as it will only be parsed by the Faraday JSON middleware if
+          # the response is content_type 'application/json'. This would of been handled automatically by the
+          # ::FaradayMiddleware::ParseJson but that is no longer available.
+          def process_body(env)
```

Comment:
Sounds like this is no longer needed because Faraday does it.

Reply:
Yeah this wasn't needed after all and has been removed.